libcudacxx/include/cuda/__iterator/discard_iterator.h

```diff
@@ -0,0 +1,343 @@
+    --__index_;
+    return __tmp;
+  }
+
+  //! @brief Returns a copy of this \c discard_iterator advanced by \p __n
+  //! @param __n The number of elements to advance
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr discard_iterator operator+(difference_type __n) const noexcept
+  {
+    return discard_iterator{__index_ + __n};
+  }
+
+  //! @brief Returns a copy of \p __x advanced by \p __n
+  //! @param __n The number of elements to advance
+  //! @param __x The original \c discard_iterator
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI friend constexpr discard_iterator
+  operator+(difference_type __n, const discard_iterator& __x) noexcept
+  {
+    return __x + __n;
+  }
+
+  //! @brief Advances the index of this \c discard_iterator by \p __n
+  //! @param __n The number of elements to advance
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr discard_iterator& operator+=(difference_type __n) noexcept
+  {
+    __index_ += __n;
+    return *this;
+  }
+
+  //! @brief Returns a copy of this \c discard_iterator decremented by \p __n
+  //! @param __n The number of elements to decrement
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr discard_iterator operator-(difference_type __n) const noexcept
+  {
+    return discard_iterator{__index_ - __n};
+  }
+
+  //! @brief Returns the distance between \p __lhs and \p __rhs
+  //! @param __lhs The left \c discard_iterator
+  //! @param __rhs The right \c discard_iterator
+  //! @return __rhs.__index_ - __lhs.__index_
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI friend constexpr difference_type
+  operator-(const discard_iterator& __lhs, const discard_iterator& __rhs) noexcept
+  {
+    return __rhs.__index_ - __lhs.__index_;
+  }
+
+  //! @brief Returns the distance between \p __lhs a \p default_sentinel
+  //! @param __lhs The left \c discard_iterator
+  //! @return -__lhs.__index_
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI friend constexpr difference_type
+  operator-(const discard_iterator& __lhs, _CUDA_VSTD::default_sentinel_t) noexcept
+  {
+    return static_cast<difference_type>(-__lhs.__index_);
+  }
+
+  //! @brief Returns the distance between a \p default_sentinel and \p __rhs
+  //! @param __rhs The right \c discard_iterator
+  //! @return __rhs.__index_
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI friend constexpr difference_type
+  operator-(_CUDA_VSTD::default_sentinel_t, const discard_iterator& __rhs) noexcept
+  {
+    return static_cast<difference_type>(__rhs.__index_);
+  }
+
+  //! @brief Decrements the index of the \c discard_iterator by \p __n
+  //! @param __n The number of elements to decrement
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr discard_iterator& operator-=(difference_type __n) noexcept
+  {
+    __index_ -= __n;
+    return *this;
+  }
+
+  //! @brief Compares two \c discard_iterator \p __lhs and \p __rhs for equality
+  //! @param __lhs The left \c discard_iterator
+  //! @param __rhs The right \c discard_iterator
+  //! @return true if both iterators store the same index
```

Comment:
Suggestions: the comments for the comparison operators seem too verbose for me. Maybe just simplify them like:
```suggestion
  //! @brief Compares two \c discard_iterator for equality by comparing their indices
```

Reply:
Shortended them